editor/editor_inspector.cpp

```diff
@@ -2570,15 +2573,41 @@ void EditorInspector::update_tree() {
 		// Get the property label's string.
 		String property_label_string = (path.find("/") != -1) ? path.substr(path.rfind("/") + 1) : path;
 		if (capitalize_paths) {
-			// Capitalize paths.
-			int dot = property_label_string.find(".");
-			if (dot != -1) {
-				String ov = property_label_string.substr(dot);
-				property_label_string = property_label_string.substr(0, dot);
-				property_label_string = property_label_string.capitalize();
-				property_label_string += ov;
+			// Capitalize a script variable only if it does not clash with any other script variable after capitalization
+			if (p.usage & PROPERTY_USAGE_SCRIPT_VARIABLE) {
+				// Create a frequency map of the capitalized script variables to check for clashes
+				if (script_vars_remaining == 0) {
+					List<PropertyInfo>::Element *E = E_property;
+					capitalized_script_vars_freq.clear();
```

Comment:
There is no need to keep frequency for variables. Simply capitalize a variable if it wasn't already added to the set.